src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -2552,101 +2545,35 @@ protected internal override Expression VisitRuntimeVariables(RuntimeVariablesExp
 
             private static IRuntimeVariables MergeRuntimeVariables(IRuntimeVariables first, IRuntimeVariables second, int[] indexes)
             {
-                return new MergedRuntimeVariables(first, second, indexes);
+                return new RuntimeOps.MergedRuntimeVariables(first, second, indexes);
             }
 
             protected internal override Expression VisitParameter(ParameterExpression node)
             {
-                LocalVariable var;
-                if (_variables.TryGetValue(node, out var))
+                IStrongBox box = GetBox(node);
+                if (box == null)
                 {
-                    return Expression.Convert(
-                        Expression.Field(
-                            var.LoadFromArray(
```

Comment:
Ideally, the shape of the tree produced by evaluation of `Quote` should be identical across compiler and interpreter because the receiver of the `Expression` object can be relying on its shape.

This said, I'd like to make sure we can run these changes on an AOT target to make sure we're not regressing anything here.